core/flyout.js

```diff
@@ -684,6 +720,44 @@ Blockly.Flyout.prototype.layoutBlocks_ = function(blocks, gaps) {
 };
 
 /**
+ * Create and place a checkbox corresponding to the given block.
+ * @param {!Blockly.Block} block The block to associate the checkbox to.
+ * @param {number} cursorX The x position of the cursor during this layout pass.
+ * @param {number} cursorY The y position of the cursor during this layout pass.
+ * @param {!{height: number, width: number}} blockHW The height and width of the
+ *     block.
+ * @param {Element} root The svg root of the block.
```

Comment:
GetHeightWidth() is not cheap--it walks the tree of blocks.  The results are needed here and in the caller function.

root has a simple getter.  I can get it in here instead.